File: frontend/src/types/lookingGlass.ts
export interface LookingGlassQuery {
  id?: string;
  type: 'ping' | 'traceroute' | 'bgp' | 'bgp-summary';
  target: string;
  router: string;
  timestamp: Date;
  status: 'pending' | 'running' | 'completed' | 'error';
  output?: string;
  error?: string;
}

export interface Router {
  id: number;
  name: string;
  hostname: string;
  location: string;
  status: 'online' | 'offline';
  ip_origens?: Array<{
    id: number;
    name: string;
    type: string;
    ip: string;
  }>;
}

export interface QueryRequest {
  type: 'ping' | 'traceroute' | 'bgp' | 'bgp-summary';
  target: string;
  routerId: number;
  options?: {
    count?: number;
    timeout?: number;
    maxHops?: number; // Para traceroute
    sourceIp?: string; // IP de origem para ping/traceroute
    ipVersion?: 4 | 6; // Versão do IP
  };
}

export interface QueryResponse {
  id: string;
  status: 'success' | 'error';
  data?: string;
  error?: string;
  executionTime?: number;
}

export const QUERY_TYPES = [
  { value: 'ping', label: 'Ping' },
  { value: 'traceroute', label: 'Traceroute' },
  { value: 'bgp', label: 'BGP Lookup' },
  { value: 'bgp-summary', label: 'BGP Lookup (Resumido)' }
] as const;

File: backend/app/schemas/looking_glass.py
from pydantic import BaseModel
from typing import Optional, List, Literal, Union
from datetime import datetime

class QueryRequest(BaseModel):
    type: Literal["ping", "traceroute", "bgp", "bgp-summary"]
    target: str
    routerId: int  # Frontend envia routerId, não router_id
    options: Optional[dict] = {}

class QueryResponse(BaseModel):
    id: str
    status: Literal["success", "error"]
    data: Optional[str] = None
    error: Optional[str] = None
    execution_time: Optional[float] = None

class LookingGlassQuery(BaseModel):
    id: Optional[str] = None
    type: Literal["ping", "traceroute", "bgp", "bgp-summary"]
    target: str
    router: str
    timestamp: datetime
    status: Literal["pending", "running", "completed", "error"]
    output: Optional[str] = None
    error: Optional[str] = None

class IpOrigem(BaseModel):
    id: int
    name: str
    type: str
    ip: str

class RouterInfo(BaseModel):
    id: int
    name: str
    hostname: str
    location: str
    status: Literal["online", "offline"]
    ip_origens: Optional[List[IpOrigem]] = []
